test/onnx/test_pytorch_onnx_onnxruntime.py

```diff
@@ -9246,6 +9246,103 @@ def forward(self, x):
         x = torch.ones(0)
         self.run_test(M(), (x,))
 
+    @skipIfUnsupportedMinOpsetVersion(11)
+    def test_broad_cast_tensors(self):
+        class M(torch.nn.Module):
+            def forward(self, x, y):
+                m = torch.broadcast_tensors(x, y)
+                return m
+
+        x = torch.randint(5, (1,))
+        y = torch.randint(5, (5,))
+
+        self.run_test(M(), (x, y))
+
+        x = torch.randint(5, (4, 2, 1, 4))
+        y = torch.randint(5, (2, 3, 1))
+
+        self.run_test(M(), (x, y))
+
+        x = torch.randn(2, 1, 4)
+        y = torch.randn(5, 2, 3, 1)
+
+        self.run_test(M(), (x, y))
+
+    @disableScriptTest()
+    @skipIfUnsupportedMinOpsetVersion(11)
+    def test_dist_normal(self):
+        class M(torch.nn.Module):
+            def forward(self, x, y):
+                return torch.distributions.Normal(x, y).sample().size(0), x, y
+
+        self.run_test(M(), (torch.tensor([0.0]), torch.tensor([[1.0], [2.0]])))
+        self.run_test(M(), (torch.tensor([0.0]), torch.tensor([1.0])))
+
+        self.run_test(M(), (torch.tensor([[[0.0], [10.0]], [[2.0], [8.0]], [[2.0], [8.0]]]), torch.tensor([[1.0], [3.0]])))
+
+    @disableScriptTest()
+    @skipIfUnsupportedMinOpsetVersion(11)
+    def test_dist_normal_correctness(self):
+        class M(torch.nn.Module):
+            def forward(self, x, y):
+                return torch.distributions.Normal(x, y).sample()
+
+        expected_mean = 5.0
+        expected_var = 10.0
+
+        model_export = M()
+        dummy_input = (torch.tensor([expected_mean]), torch.tensor([expected_var]))
+        ort_sess = convert_to_onnx(model_export, input=dummy_input, opset_version=self.opset_version,
+                                   training=torch.onnx.TrainingMode.EVAL)
+
+        ort_out = []
+        for i in range(1500):
+            ort_out.append(run_ort(ort_sess, input=dummy_input))
+
+        actual_var = np.var(ort_out)
+        actual_mean = np.mean(ort_out)
+
+        assert abs(abs(actual_mean) - expected_mean) <= expected_mean * 0.15, \
```

Comment:
Why are you using the 0.15 constant?

Reply:
After some tests, 0.15 is a stable offset between our PyTorch value and ONNX value for many rounds of running this test. And I also think it is an acceptable one for the accuracy validation.

Reply:
The expected mean also should follow the normal distribution, with (mean, variance/sample_size). Using the interval calculator for normal distribution https://keisan.casio.com/exec/system/14060744929691, the probability of passing the test is only 99.6%. On average it takes ~250 runs to produce a failure case. One suggestion would be to increase the sample size from 1500 to 10000, which guarantees 0.99999999999994 success rate. 

np.var is computing variance, however variable `expected_var` should actually be `expected_std`, since that's the input `torch.distributions.Normal` take. And with the redundant sqrt in the symbolic, although the exported graph is not the same with pytorch, the test assertion actually passed... For sample variance, it is following chi square distribution with degree of freedom of sample_size - 1. Using sample size of 10000 should also provide good success rate for the check.

And to increase performance of the test, it is recommended to generate 10000 samples directly in the model `torch.distributions.Normal(x, y).sample([10000])`. This should still be fast enough.